### ISSUE TITLE
Crash in raid

### ISSUE THREAD

**Issue by BRNSystems** [Patbox/IllagerExpansionRecrafted#5]
```
[17:01:16] [Server thread/INFO]: Async chunk loading for player BRNSystems completed
[17:01:16] [Server thread/INFO]: [ShulkerBoxTooltip] BRNSystems: protocol version: 2.0
[17:01:16] [Server thread/INFO]: [ShulkerBoxTooltip Plugins] Loading 1 plugin: shulkerboxtooltip
[17:01:16] [Server thread/INFO]: [ShulkerBoxTooltip] overriding preview provider shulkerboxtooltip:trapped_chest with shulkerboxtooltip:chest for item minecraft:chest
[17:01:16] [Server thread/INFO]: [ShulkerBoxTooltip Plugins] Registered 31 providers for mod shulkerboxtooltip
[17:01:16] [Server thread/INFO]: Player BRNSystems joined with a matching carpet client
[17:01:16] [VoiceChatPacketProcessingThread/INFO]: [voicechat] Successfully authenticated player c4240be5-4ecf-4d83-9e0e-c00ff562306c
[17:01:17] [VoiceChatPacketProcessingThread/INFO]: [voicechat] Successfully validated connection of player c4240be5-4ecf-4d83-9e0e-c00ff562306c
[17:01:17] [VoiceChatPacketProcessingThread/INFO]: [voicechat] Player BRNSystems (c4240be5-4ecf-4d83-9e0e-c00ff562306c) successfully connected to voice chat
[17:01:17] [Netty Epoll Server Play IO #0/ERROR]: Error sending packet clientbound/minecraft:set_entity_data
io.netty.handler.codec.EncoderException: Failed to encode packet 'clientbound/minecraft:set_entity_data'
        at net.minecraft.class_9136.method_56426(class_9136.java:55) ~[server-intermediary.jar:?]
        at net.minecraft.class_9136.encode(class_9136.java:14) ~[server-intermediary.jar:?]
        at net.minecraft.class_2545.method_10838(class_2545.java:26) ~[server-intermediary.jar:?]
        at net.minecraft.class_2545.encode(class_2545.java:12) ~[server-intermediary.jar:?]
        at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:107) ~[netty-codec-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:113) ~[netty-codec-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.ChannelOutboundHandlerAdapter.write(ChannelOutboundHandlerAdapter.java:113) ~[netty-transport-4.1.97.Final.jar:?]
        at net.minecraft.class_2535$2.write(class_2535.java:518) ~[server-intermediary.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:851) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1010) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannel.write(AbstractChannel.java:296) ~[netty-transport-4.1.97.Final.jar:?]
        at net.minecraft.class_2535.method_36942(class_2535.java:342) ~[server-intermediary.jar:?]
        at net.minecraft.class_2535.method_52917(class_2535.java:337) ~[server-intermediary.jar:?]
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) ~[netty-common-4.1.97.Final.jar:?]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) ~[netty-common-4.1.97.Final.jar:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[netty-common-4.1.97.Final.jar:?]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:413) ~[netty-transport-classes-epoll-4.1.97.Final.jar:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.97.Final.jar:?]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.97.Final.jar:?]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.97.Final.jar:?]
        at java.base/java.lang.Thread.run(Thread.java:1570) [?:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 17 out of bounds for length 16
        at net.minecraft.class_2739.polymer$createEntries(class_2739.java:1071) ~[server-intermediary.jar:?]
        at net.minecraft.class_2739.modify$dan000$polymer-core$polymer$changeForPacket(class_2739.java:1105) ~[server-intermediary.jar:?]
        at net.minecraft.class_2739.method_55926(class_2739.java:44) ~[server-intermediary.jar:?]
        at net.minecraft.class_9139$6.encode(class_9139.java:42) ~[server-intermediary.jar:?]
        at net.minecraft.class_9139$9.method_56442(class_9139.java:97) ~[server-intermediary.jar:?]
        at net.minecraft.class_9139$9.encode(class_9139.java:87) ~[server-intermediary.jar:?]
        at net.minecraft.class_9136.method_56426(class_9136.java:53) ~[server-intermediary.jar:?]
        ... 32 more
[17:01:17] [Server thread/INFO]: BRNSystems lost connection: Internal Exception: io.netty.handler.codec.EncoderException: Failed to encode packet 'clientbound/minecraft:set_entity_data'
[17:01:17] [Server thread/INFO]: BRNSystems nás opustil
[17:01:17] [Server thread/INFO]: [voicechat] Disconnecting client BRNSystems
[17:01:19] [Server thread/INFO]: Arrgy[/178.41.23.20:50852] logged in with entity id 259 at (573.2405483336008, 74.41391426255468, -3800.287864790469)
[17:01:19] [Server thread/INFO]: Full sync for Arrgy took 98.46 ms
[17:01:19] [Server thread/INFO]: Vitaj späť, 'Arrgy'
[17:01:19] [Server thread/INFO]: Async chunk loading for player Arrgy completed
[17:01:19] [Server thread/INFO]: [voicechat] Received secret request of Arrgy (18)
[17:01:19] [Server thread/INFO]: [voicechat] Sent secret to Arrgy
[17:01:19] [Netty Epoll Server Play IO #1/ERROR]: Error sending packet clientbound/minecraft:set_entity_data
io.netty.handler.codec.EncoderException: Failed to encode packet 'clientbound/minecraft:set_entity_data'
        at net.minecraft.class_9136.method_56426(class_9136.java:55) ~[server-intermediary.jar:?]
        at net.minecraft.class_9136.encode(class_9136.java:14) ~[server-intermediary.jar:?]
        at net.minecraft.class_2545.method_10838(class_2545.java:26) ~[server-intermediary.jar:?]
        at net.minecraft.class_2545.encode(class_2545.java:12) ~[server-intermediary.jar:?]
        at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:107) ~[netty-codec-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:851) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.handler.codec.MessageToMessageEncoder.writePromiseCombiner(MessageToMessageEncoder.java:140) ~[netty-codec-4.1.97.Final.jar:?]
        at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:120) ~[netty-codec-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.ChannelOutboundHandlerAdapter.write(ChannelOutboundHandlerAdapter.java:113) ~[netty-transport-4.1.97.Final.jar:?]
        at net.minecraft.class_2535$2.write(class_2535.java:518) ~[server-intermediary.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:851) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1010) ~[netty-transport-4.1.97.Final.jar:?]
        at io.netty.channel.AbstractChannel.write(AbstractChannel.java:296) ~[netty-transport-4.1.97.Final.jar:?]
        at net.minecraft.class_2535.method_36942(class_2535.java:342) ~[server-intermediary.jar:?]
        at net.minecraft.class_2535.method_52917(class_2535.java:337) ~[server-intermediary.jar:?]
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) ~[netty-common-4.1.97.Final.jar:?]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) ~[netty-common-4.1.97.Final.jar:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[netty-common-4.1.97.Final.jar:?]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:416) ~[netty-transport-classes-epoll-4.1.97.Final.jar:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.97.Final.jar:?]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.97.Final.jar:?]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.97.Final.jar:?]
        at java.base/java.lang.Thread.run(Thread.java:1570) [?:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 17 out of bounds for length 16
        at net.minecraft.class_2739.polymer$createEntries(class_2739.java:1071) ~[server-intermediary.jar:?]
        at net.minecraft.class_2739.modify$dan000$polymer-core$polymer$changeForPacket(class_2739.java:1105) ~[server-intermediary.jar:?]
        at net.minecraft.class_2739.method_55926(class_2739.java:44) ~[server-intermediary.jar:?]
        at net.minecraft.class_9139$6.encode(class_9139.java:42) ~[server-intermediary.jar:?]
        at net.minecraft.class_9139$9.method_56442(class_9139.java:97) ~[server-intermediary.jar:?]
        at net.minecraft.class_9139$9.encode(class_9139.java:87) ~[server-intermediary.jar:?]
        at net.minecraft.class_9136.method_56426(class_9136.java:53) ~[server-intermediary.jar:?]
        ... 34 more
```

**Comment by Patbox** [comment]
Hey, does it happen with latest polymer?

**Comment by BRNSystems** [comment]
yes, 0.8.2

**Comment by Fantomrat** [comment]
I get the same error when invoker casts a spell to summon surrendered

- fabric-api-0.100.0+1.20.6
- illager_expansion-1.2.0+1.20.6
- polymer-bundled-0.8.3+1.20.6
- trinkets-3.9.0+polymerport.1

**Comment by Patbox** [comment]
Should be fixed by updating polymer to 0.8.4 or newer
